### PR TITLE
Adding ArtifactsRetriever ThreadFactory

### PR DESCRIPTION
**Pull request by Sarangan0219**
Related to wso2/product-apim#8246
Resolves https://github.com/wso2/carbon-apimgt/pull/9016#discussion_r458945368

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package org.wso2.carbon.apimgt.gateway.listeners;
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java:1]: Missing License Header

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This is the Thread Factory used to create the threads which retrieve the artifacts from the extension.
+ */
+public class ArtifactsRetrieverThreadFactory implements ThreadFactory {
+    static final AtomicInteger POOL_NUMBER = new AtomicInteger(1);
+    final ThreadGroup group;
+    final AtomicInteger threadNumber = new AtomicInteger(1);
```
Comment (jaadds) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java:12]: mark as private

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java (hunk 1 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java:1]: Fixed https://github.com/wso2/carbon-apimgt/pull/9541/commits/2809e66ce7558218fa0dd2ac5e74a4e82ff5fb83

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java (hunk 2 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/ArtifactsRetrieverThreadFactory.java:12]: Fixed https://github.com/wso2/carbon-apimgt/pull/9541/commits/2809e66ce7558218fa0dd2ac5e74a4e82ff5fb83

**Pull request merged**